opencti-platform/opencti-graphql/src/migrations/1716815108224-update-data-sharing-ingestion-capabilities.js

```diff
@@ -0,0 +1,79 @@
+import { logApp } from '../config/conf';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { elLoadById, elReplace } from '../database/engine';
+import { roleCapabilities } from '../domain/user';
+import { addCapability } from '../domain/grant';
+import { createRelation } from '../database/middleware';
+import { listAllEntities } from '../database/middleware-loader';
+import { ENTITY_TYPE_ROLE } from '../schema/internalObject';
+
+const message = '[MIGRATION] Split data sharing & ingestion into 2 capabilities';
+
+export const up = async (next) => {
+  logApp.info(`${message} > started`);
+  const context = executionContext('migration');
+
+  // ------ Create Access ingestion
+  const accessIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION',
+    attribute_order: 2600,
+    description: 'Access ingestion'
+  });
+  // ------ Create Manage ingestion
+  const manageIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION_SETINGESTIONS',
+    description: 'Manage ingestion',
+    attribute_order: 2610
+  });
+
+  // ------ Update roles
+  const roles = await listAllEntities(context, SYSTEM_USER, [ENTITY_TYPE_ROLE], {});
+  for (let i = 0; i < roles.length; i += 1) {
+    const roleId = roles[i].id;
+    const capabilities = await roleCapabilities(context, SYSTEM_USER, roleId);
+    // Select 'Access ingestion' if 'Access Data sharing & ingestion' is selected
+    const hasAccessDataSharingCapability = capabilities.some((capability) => capability.name === 'TAXIIAPI');
+    if (hasAccessDataSharingCapability) {
+      const input = { fromId: roleId, toId: accessIngestionCapability.id, relationship_type: 'has-capability' };
+      await createRelation(context, SYSTEM_USER, input);
+    }
+    // Select 'Manage ingestion' if 'Manage Data sharing & ingestion' is selected
+    const hasManageDataSharingCapability = capabilities.some((capability) => capability.name === 'TAXIIAPI_SETCOLLECTIONS');
+    if (hasManageDataSharingCapability) {
+      const input = { fromId: roleId, toId: manageIngestionCapability.id, relationship_type: 'has-capability' };
+      await createRelation(context, SYSTEM_USER, input);
+    }
+  }
+
+  // ------ Update Attribute_order and name of Manage CSV mappers
+  const CSVCapability = await elLoadById(context, SYSTEM_USER, 'capability--5407e8a4-0ff9-5253-89b1-c01a92ad9453');
```

Comment:
Just out of curiosity: Where this id is coming from ? (capability--5407e8a4-0ff9-5253-89b1-c01a92ad9453)

Reply:
This is the standard_id property of a capability. You can find it like this :
```
{
  capabilities{
    edges{
      node{
        name
        description
        attribute_order
        standard_id
      }
    }
  }
}
```
I reproduce the process of a previous migration using standard_id in this File : `1713449762000-investigations-and-dashboards-capabilities`

Reply:
Ok thanks. I thought those ids were proper for each platform and a different one was generated for you and me for example, so thta's why I was confused.